src/api.c

```diff
@@ -211,14 +223,25 @@ gtk_layer_auto_exclusive_zone_is_enabled (GtkWindow *window)
 }
 
 void
-gtk_layer_set_keyboard_interactivity (GtkWindow *window, gboolean interacitvity)
+gtk_layer_set_keyboard_interactivity (GtkWindow *window, GtkLayerShellKeyboardInteractivity interactivity)
 {
     LayerSurface *layer_surface = gtk_window_get_layer_surface (window);
     if (!layer_surface) return; // Error message already shown in gtk_window_get_layer_surface
-    layer_surface_set_keyboard_interactivity (layer_surface, interacitvity);
+    uint32_t version = gtk_layer_get_layer_shell_version ();
+    if (interactivity >= GTK_LAYER_SHELL_KEYBOARD_ENTRY_NUMBER)
+    {
+        g_warning ("Invalid value for keyboard interactivity setting!\n");
+        return;
+    }
+    if (version <= 3 && interactivity == GTK_LAYER_SHELL_KEYBOARD_ON_DEMAND)
```

Comment:
Might be a little cleaner to make this check in layer-surface.c. I think getting the version from the `zwlr_layer_surface_v1` (which you'll have access to in that file) instead of from the layer shell global would be slightly preferable.

Reply:
`zwlr_layer_surface_v1_get_version` can be used for this purpose.

Reply:
Moved the check there. The problem is that `self->layer_surface` can be NULL when this is called, so in that case, we still have to use the global object as fallback (I think it is cleaner to have the warning here and not have to make this check when committing the surface later).